qemu/qemu.go

```diff
@@ -1504,6 +1526,22 @@ func (config *Config) appendBios() {
 	}
 }
 
+func (config *Config) appendMigrateIncoming() {
+	if config.Incoming.URI != "" {
+		var proto string
+		var fd int
+		uri := config.Incoming.URI
+		n, err := fmt.Sscanf(uri, "%s:%d", &proto, &fd)
+		if err == nil && n == 2 && proto == "fd" && fd > 2 {
+			chFDs := config.appendFDs([]*os.File{os.NewFile(uintptr(fd), "fd incoming")})
```

Comment:
Is the intention here that the calling process has already opened the file whose descriptor is given in the URI?  If so I'm not sure about this.  In other parts of the API will expect the caller to open the file and then pass in the *os.File object.